openpype/hosts/photoshop/plugins/create/workfile_creator.py

```diff
@@ -17,7 +16,7 @@ def get_instance_attr_defs(self):
         return []
 
     def collect_instances(self):
-        for instance_data in api.list_instances():
+        for instance_data in self.host.list_instances():
```

Comment:
I would recommend using shared data if instances are stored at the same place (shared data across creators during collection phase). Check TrayPublisher for inspiration https://github.com/pypeclub/OpenPype/blob/develop/openpype/hosts/traypublisher/api/plugin.py#L20 .

Reply:
Why, `self.host` gets reinitialized?

Or you mean to cache value of  `list_instances`? I could create internal `_instances` list and add it to `list_instances` method.

Reply:
Attribute `collection_shared_data` is shared dictionary across all creators so they don't have to collect exactly the same data individually.